src/usage.rs

```diff
@@ -0,0 +1,23 @@
+//! The module for the usage advice.
```

Comment:
It was intended to test the capabilities of the module system and the docstrings at once without uglifying the main source file.

It could be renamed to `utilities.rs` as a storage for utility functions.

Reply:
Right now there not really any utility functions, so I would rather not. I think what you wrote down below with putting the types into their own module makes sense. But I would maybe recommend to you, to first implement the support for the `stdin` input, since this will likely not require a seperate module. After that you have a better overview how big the project actually became, and in what way modules can help or maybe be a hindrance to readabiltiy. 😄

Reply:
At the moment, there are four standalone functions in the project:

* `main`
* `matches_filter`
* `parse_commit`
* `usage

From those four, I would just consider `main` and `parse_commit` actual business logic and the other two, `matches_filter` and `usage`, merely syntactic sugar.  In C, I would `inline` `matches_filter` (because a macro might suffer from side effects which could cause different results thereby) and turn `usage` into a macro (or `inline` it, as well).

I agree with you that an own module for a syntactic sugar function might be too much.  But my suggestion is to create one common module for such simple functions which only exist to abbreviate something.

In summary, I would like to suggest to move `matches_filter` into the `usage` module, as well, and to rename that module appropriately.

Reply:
Hmm, we could make `matches_filter` a member method in Commit. So basically something like this:

```rust
impl Commit {
    pub fn matches_filter(&self, filter: &Filter) -> bool {
        // Old code here
    }
}
```

Because I would create modules rather topic based. So that, e.g., in a module there are all functions regarding the frontend, or parsing or something like this. This makes it easier to find relevant functions, types, and also modules, for the current source file. Meanwhile in your utilities functions approach there would be a lot of functions, that would be called from something completely else, and functions, that are written close by do not have to have a correlation with each other in any way. 😅 

You get, where I'm getting at?

Reply:
I see your point.  But for this Pull Request, shall I revert the module?  If so, what about its docstrings?

Reply:
Hmm, I think, a lot of the stuff written in it, is to be expected by anyone already. Maybe you can combine it with the comment for the function, but if so, please shorten it by a lot. 😄

Reply:
Yes, I think, that is the right course of action 👍

Reply:
Done.